cpg_workflows/jobs/gcnv.py

```diff
@@ -359,3 +359,91 @@ def fix_intervals_vcf(
 
     return reheader_job
 
+
+def merge_calls(
+    b: hb.Batch, sg_vcfs: list[str], docker_image: str, job_attrs: dict[str, str], output_path: Path
+):
+    """
+    This job will run a fast simple merge on per-SGID call files
+    It then throws in a python script to add in two additional header lines
+    and edit the SVLEN and SVTYPE attributes into each row
+
+    Args:
+        b (batch):
+        sg_vcfs (list[str]): paths to all individual VCFs
+        docker_image (str): docker image to use
+        job_attrs (dict): any params to atach to the job
+        output_path (Path): path to the final merged VCF
+    """
+
+    if can_reuse(output_path):
+        return None
+
+    assert sg_vcfs, 'No VCFs to merge'
+
+    j = b.new_job('Merge gCNV calls', job_attrs | {'tool': 'bcftools'})
+    j.image(docker_image)
+
+    # this should be made reactive, in case we scale past 10GB
+    j.storage('10Gi')
+
+    batch_vcfs = []
+    for each_vcf in sg_vcfs:
+        batch_vcfs.append(
+            b.read_input_group(
+                **{
+                    'vcf.gz': each_vcf,
+                    'vcf.gz.tbi': f'{each_vcf}.tbi',
+                }
+            )['vcf.gz']
+        )
+
+    j.declare_resource_group(
+        output={'vcf.bgz': '{root}.vcf.bgz', 'vcf.bgz.tbi': '{root}.vcf.bgz.tbi'}
+    )
+
+    # option breakdown:
+    # -Oz: bgzip output
+    # -o: output file
+    # --threads: number of threads to use
+    # -m: merge strategy
+    # -0: compression level
+    j.command(
+        f'bcftools merge {" ".join(batch_vcfs)} -Oz -o temp.vcf.bgz --threads 4 -m all -0'
+    )
+    j.command(fr"""
```

Comment:
nit: Cant say I am in love with inline python.

Reply:
I really dislike this too. I think there should be a fairly simple way of doing this as a Python job followed by a tabix job, instead of bundling both operations into one Job. Having to parse the VCF and shove in extra header lines etc. it awful at the best of times.

At least I can say that this design is consistent with the GATK-SV codebase 🙃